finance-admin-api/db/bad_debt_write_off.go

```diff
@@ -0,0 +1,83 @@
+package db
+
+import (
+	"github.com/ministryofjustice/opg-sirius-supervision-finance-admin/shared"
+	"os"
+	"time"
+)
+
+type BadDebtWriteOff struct {
+	FromDate *shared.Date
+	ToDate   *shared.Date
+}
+
+const BadDebtWriteOffQuery = `SELECT CONCAT(p.firstname, ' ', p.surname)        "Customer Name",
+       p.caserecnumber                            "Customer number",
+       fc.sop_number                              "SOP number",
+       '="0470"'                                  AS "Entity",
+       cc.code                                 AS "Cost centre",
+       ac.code                                  AS "Account code",
+       ac.account_code_description              AS "Account code description",
+       ((la.amount / 100.0)::NUMERIC(10, 2))::varchar(255)       AS "Adjustment amount",
+       l.datetime                              AS "Adjustment date",
+       CASE
+           WHEN l.type = 'CREDIT WRITE OFF' THEN CONCAT('WO', i.reference)
+           ELSE CONCAT('WOR', i.reference) END AS "Txn number",
+       CONCAT(a.name, ' ', a.surname)          AS "Approver"
+FROM supervision_finance.finance_client fc
+         JOIN public.persons p ON fc.client_id = p.id
+         JOIN supervision_finance.ledger l ON fc.id = l.finance_client_id
+         JOIN supervision_finance.ledger_allocation la
+              ON l.id = la.ledger_id AND la.status = 'ALLOCATED' -- to not include unapply/reapply
+         JOIN supervision_finance.invoice i ON la.invoice_id = i.id
+         LEFT JOIN public.assignees a ON l.created_by = a.id
+         LEFT JOIN LATERAL (
+    SELECT ifr.supervisionlevel AS supervision_level
```

Comment:
This needs to change to the following to include guardianships and AD fees
`SELECT CASE
            WHEN i.feetype = 'AD' THEN 'AD'
            ELSE COALESCE(sl.supervision_level, '')
            END AS supervision_level`